console-subscriber/src/builder.rs

```diff
@@ -101,4 +104,41 @@ impl Builder {
     pub fn build(self) -> (TasksLayer, Server) {
         TasksLayer::build(self)
     }
+
+    /// Configures this builder from a standard set of environment variables:
+    ///
+    /// * `TOKIO_CONSOLE_RETENTION`: The number of seconds to accumulate
+    ///   completed tracing data. Default: 60s.
+    /// * `TOKIO_CONSOLE_BIND`: a HOST:PORT description, such as
+    ///   localhost:1234 or similar. Default: 127.0.0.1:6669
+    /// * `TOKIO_CONSOLE_PUBLISH_INTERVAL`: The number of milliseconds to wait
+    ///   between sending updates to the console. Default: 1000ms (1s)
+
+    pub fn from_default_env(mut self) -> Self {
+        if let Ok(retention) = std::env::var("TOKIO_CONSOLE_RETENTION") {
+            self.retention = Duration::from_secs(
+                retention
+                    .parse()
+                    .expect("TOKIO_CONSOLE_RETENTION must be an integer"),
+            );
+        }
+
+        if let Ok(bind) = std::env::var("TOKIO_CONSOLE_BIND") {
+            self.server_addr = bind
+                .to_socket_addrs()
+                .expect("TOKIO_CONSOLE_BIND must be formatted as HOST:PORT, such as localhost:4321")
+                .next()
+                .expect("tokio console could not resolve TOKIO_CONSOLE_BIND");
+        }
+
+        if let Ok(interval) = std::env::var("TOKIO_CONSOLE_PUBLISH_INTERVAL") {
+            self.publish_interval = Duration::from_millis(
+                interval
+                    .parse()
+                    .expect("TOKIO_CONSOLE_PUBLISH_INTERVAL must be an integer"),
+            );
+        }
```

Comment:
...and similarly, i'd change this to
```suggestion
        if let Ok(interval) = std::env::var("TOKIO_CONSOLE_PUBLISH_INTERVAL_MS") {
            self.publish_interval = Duration::from_millis(
                interval
                    .parse()
                    .expect("TOKIO_CONSOLE_PUBLISH_INTERVAL_MS must be an integer"),
            );
        }
```

Reply:
Sweet, this was also my first instinct but wasn't sure if the variable names were getting too long. Self-describing units are great 👍